language/changes/6-phantom-type-params.md

```diff
@@ -0,0 +1,103 @@
+# Phantom Type Parameters
+
+- Status: Implemented in Move 1.4
+
+## Introduction
+
+A phantom type parameter is one that doesn't show up at runtime, but is checked statically at
+compile time. Previously, Move's type system didn't make a difference on whether a type parameter
+was phantom, but with this new feature, a struct's type parameter can be explicitly declared as
+phantom. A parameter declared as phantom is not considered when computing the
+[conditional abilities with generic types](3-abilities.md#Conditional-Abilities-with-Generic-Types).
+For this relaxed rule to be sound, Move's type system guarantees that a parameter declared as
+phantom is either not used at all in the struct definition, or it is only used as an argument to
+type parameters also declared as phantom.
+
+## Motivations
+
+A phantom type parameter may come in handy in a couple of situations. For example, the struct
+`Event::EventHandle<T>` in stdlib doesn't contain any field of type `T`, but uses the type parameter
+to ensure at the type level that the handle can only be used for messages of type `T`. Another
+notorious example is `Diem::Diem<CoinType>`, which is generic on a `CoinType` specifying the
+currency of the coin and allowing code to be written generically on any currency. As a last example,
+the capabilities in `Vault` also rely heavily on phantom type parameters.
+
+Previously, these examples required spurious ability annotations to satisfy the requirements of the
+abilities declared for each of the structs. For example, `Diem::Diem<CoinType>` is declared with the
+ability `key`, which required all arguments to `CoinType` to be declared with `store`. Specifically,
+the types `XUS::XUS` and `XDX::XDX` were forced to have a `store` annotation even though they were
+never stored in the database! More over the extra `store` requirement polluted the entire Diem
+Framework and many functions had to specify an extra `store` constraints on their type parameters.
+With the new feature, `Diem::Diem<CoinType>` now declares `CoinType` to be phantom which avoids the
```

Comment:
It would be better to first state this generally, i.e., saying that this language change avoids the need for spurious abilities, before giving the specific example of `Diem<CoinType>`. It might also be good to add something before this concluding sentence to elaborate on why the spurious abilities are bad.